spec/services/submit_reference_spec.rb

```diff
@@ -2,79 +2,101 @@
 
 RSpec.describe SubmitReference do
   describe '#save!' do
-    let(:application_form) { create(:completed_application_form) }
+    describe 'decoupled_references feature ON' do
```

Comment:
This diff looks noisy, but all that's happened here is I've introduced two separate `describe` blocks for `decoupled_references feature ON` and `decoupled_references feature OFF`. A new example sits in the former, all the old examples are nested under the latter.

Reply:
Looks fine with "ignore whitespace"!

Reply:
It's not overly important, but out of interest I would have put the flag being on and off in context blocks. What's the thought process of using describe? I'm mainly interested as I think I've been doing it wrong.

Reply:
@davidgisbey completely stylistic, these could readily have been context blocks. I guess here I was thinking that the method behaviour is almost completely different with the flag active, rather than the minor variation that a context probably suggests.